client/src/CoqDocument.ts

```diff
@@ -298,16 +299,17 @@ export class CoqDocument implements vscode.Disposable {
       }
       let stat : fs.Stats|undefined = undefined;
       try {
-        stat = fs.statSync(path.join(v, 'coqtop'));
-      } catch(err) {}
-      if(!stat && os.platform()==='win32') {
-        try {
-          stat = fs.statSync(path.join(v, 'coqtop.exe'));
-        } catch(err) { }
+        stat = fs.statSync(path.join(v, coqtopExe));
+      } catch {
+        if (os.platform() === 'win32') {
+          try {
+            stat = fs.statSync(path.join(v, 'coqtop.exe'));
```

Comment:
Argh, seems my comments got lost. In short, this should probably use `coqtopExe + '.exe'` (tho that's arguable), and moving this fallback into the exception handler seems dangerous without sufficient testing (does stat only fail through exceptions, on all platforms? Is that actually guaranteed?) — it doesn't belong to the same PR at any rate.

Reply:
See [statSync on node](https://nodejs.org/api/fs.html#fs_fs_statsync_path_options): the only failure mode is via an exception.

And no, it shouldn't be `coqtopExe` + '.exe'. We expect users to specify `coqtopExe` with a '.exe': I updated this patch mainly for backward compatibility; it's for Windows users who want VSCoq to work right out of the box, when 'coqtop.exe' is in `$PATH`.

Reply:
> We expect users to specify coqtopExe with a '.exe': I updated this patch mainly for backward compatibility; 

That’s not written in the setting description tho, is it?

Reply:
Good point. Fixed.